src/vehicle.h

```diff
@@ -233,6 +233,17 @@ enum class vp_flag : uint32_t {
     tracked_flag = 16 //carried vehicle part with tracking enabled
 };
 
+class turret_cpu
+{
+    public:
+        std::unique_ptr<npc> brain;
+        turret_cpu() = default;
+        turret_cpu( const turret_cpu & ) {};
+        turret_cpu &operator=( const turret_cpu & ) {
+            return *this;
```

Comment:
```suggestion
            brain.reset();
            return *this;
```
If the copy assignment function is called then presumably its vehicle part is being assigned with a different value, so the NPC should be invalidated.

I also suggest changing `turret_cpu` to be a private inner class of `vehicle_part`, making `turret_cpu cpu` private, and changing `get_targeting_npc` to be a member function of `vehicle_part` that takes a `vehicle` parameter instead, to keep `turret_cpu` as an implementation detail of `vehicle_part`.

Reply:
Thanks for your suggestions. I think `turret_cpu` has already been private in `vehicle_part` ( the default access modifier ).